### PR TITLE
added function for converting between per m and per hz quantities.

### PR DESCRIPTION
**Pull request by j-petersen**
added function for converting between per wavelength and per frequency quantities.

### CODE REVIEW

File: typhon/physics/em.py [hunk 1]
```diff
@@ -349,6 +351,44 @@ def wavenumber2wavelength(wavenumber):
...
+
+
+def perwavelength2perfrequency(perm, lam_grid):
+    """Conversion between Wavelength- and Frequency-related Spectral Quantities.
+
+    Note that the array is still sorted by wavelength, and therefore by decreasing frequency.
+
+    Parameters:
+        pernm (ndarray)[nlam_grid, ...]: Quantity per wavelength [?/m].
+        f_grid (1darray)[nlam_grid]: Wavelength [m].
```
Comment (olemke) [inline typhon/physics/em.py]: docstring parameter name doesn't match function parameter name. `f_grid` should be `lam_grid`.

File: typhon/physics/em.py [hunk 2]
```diff
@@ -349,6 +351,44 @@ def wavenumber2wavelength(wavenumber):
...
+    return perm
+
+
+def perwavelength2perfrequency(perm, lam_grid):
+    """Conversion between Wavelength- and Frequency-related Spectral Quantities.
+
+    Note that the array is still sorted by wavelength, and therefore by decreasing frequency.
+
+    Parameters:
+        pernm (ndarray)[nlam_grid, ...]: Quantity per wavelength [?/m].
```
Comment (olemke) [inline typhon/physics/em.py]: Typo `pernm` -> `perm`

**Comment by lkluft** [comment]
I was wondering if the functions should also return the converted grid, i.e.:
```py
perm, m = perfrequency2perwavelength(irrad, f_grid)
```
I would argue that users most likely need both quantities for further processing (plotting, integration, ...).

In addition, this would allow to add a back-and-forth test:
```py
f_grid = np.array([4e14, 6e14, 8e14])
irrad = np.array([1e-12, 2e-12, 3e-12])

i_perm, lam_grid = physics.perfrequency2perwavelength(irrad, f_grid)
i_perf, f_out = physics.perwavelength2perfrequency(i_perm, lam_grid)

assert np.allclose(x, ref)
assert np.allclose(f_out, f_grid)
```

**Comment by j-petersen** [comment]
@lkluft Yes, you are right. I added the converted grid and swapped the ordering so that it is in increasing order again.

**Pull request merged**